syntax = "proto3";

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

option go_package = "proto/topic";


message Topic {
        string name = 1;
        google.protobuf.Timestamp lastUpdated = 2;
}

message TopicResponse {
        repeated Topic topics = 1;
}

message GetPendingTopicsRequest {
        google.protobuf.Timestamp lastUpdated = 1;
}

message AddTopicsRequest {
        repeated Topic topics = 1;
}

message RemoveTopicsRequest {
        repeated string topicNames = 1;
}

service TopicService {
        rpc AddTopics(AddTopicsRequest) returns (TopicResponse);
        rpc GetTopics(google.protobuf.Empty) returns (TopicResponse);
        rpc RemoveTopics(RemoveTopicsRequest) returns (google.protobuf.Empty);
        rpc GetPendingTopics(GetPendingTopicsRequest) returns (stream TopicResponse);
}

